rrmngmnt/power_manager.py

```diff
@@ -0,0 +1,129 @@
+"""
+Manage host power via ssh or ipmitool
+"""
+import socket
+import subprocess
+from rrmngmnt.service import Service
+
+
+class PowerManager(Service):
+    """
+    Base power management class
+    """
+    reboot_command = None
+    poweron_command = None
+    poweroff_command = None
+
+    def _exec_pm_command(self, command, *args):
+        raise NotImplementedError("Must be implemented under child class")
+
+    def restart(self, *args):
+        """
+        Reboot host
+        """
+        self._exec_pm_command(self.reboot_command, *args)
+
+    def poweroff(self, *args):
+        """
+        Power off host
+        """
+        self._exec_pm_command(self.poweroff_command, *args)
+
+    def poweron(self, *args):
+        """
+        Power on host
+        """
+        self._exec_pm_command(self.poweron_command, *args)
+
+
+class SSHPowerManager(PowerManager):
+    """
+    SSH power management class
+    """
+    reboot_command = ["reboot"]
+    poweroff_command = ["poweroff"]
+
+    def _exec_pm_command(self, command, *args):
+        try:
+            command += args
```

Comment:
```
from rrmngmnt import Host
from rrmngmnt import RootUser
from rrmngmnt.power_manager import SSHPowerManager


if __name__ == "__main__":
    h = Host('10.34.63.218')
    h.users.append(RootUser('123456'))
    p = SSHPowerManager(h)
    p.restart(1)
```
and running that
```
In [1]: p
Out[1]: <rrmngmnt.power_manager.SSHPowerManager at 0x7f5746179750>

In [2]: p.reboot_command
Out[2]: ['reboot', 1]

In [3]: p.restart(1)

In [4]: p.reboot_command
Out[4]: ['reboot', 1, 1]
```
I think you can remove that *args, or handle it properly ...